api/v1_coins_volume_leaders.go

```diff
@@ -0,0 +1,168 @@
+package api
+
+import (
+	"time"
+
+	"api.audius.co/api/dbv1"
+	"github.com/gofiber/fiber/v2"
+	"github.com/jackc/pgx/v5"
+)
+
+type VolumeLeaderRow struct {
+	Address string  `json:"address" db:"address"`
+	Volume  float64 `json:"volume" db:"volume"`
+	UserID  *int32  `json:"-" db:"user_id"`
+}
+
+type VolumeLeaderUserFull struct {
+	VolumeLeaderRow
+	User *dbv1.FullUser `json:"user"`
+}
+
+type VolumeLeaderUserMin struct {
+	VolumeLeaderRow
+	User *dbv1.MinUser `json:"user"`
+}
+
+type GetCoinsVolumeLeadersQueryParams struct {
+	Limit int `query:"limit" default:"20" validate:"min=1,max=100"`
+	// Max offset of 500 to prevent expensive queries
+	Offset   int    `query:"offset" default:"0" validate:"min=0,max=500"`
+	FromDate string `query:"from" validate:"omitempty,datetime=2006-01-02T15:04:05Z07:00"`
+	ToDate   string `query:"to" validate:"omitempty,datetime=2006-01-02T15:04:05Z07:00"`
+}
+
+func (app *ApiServer) v1CoinsVolumeLeaders(c *fiber.Ctx) error {
+	queryParams := GetCoinsVolumeLeadersQueryParams{}
+	if err := app.ParseAndValidateQueryParams(c, &queryParams); err != nil {
+		return err
+	}
+
+	now := time.Now().UTC()
+
+	// Default to midnight of the current day (UTC)
+	fromDate := time.Date(now.Year(), now.Month(), now.Day(), 0, 0, 0, 0, time.UTC)
+	if queryParams.FromDate != "" {
+		parsed, err := time.Parse(time.RFC3339, queryParams.FromDate)
+		if err != nil {
+			return fiber.NewError(fiber.StatusBadRequest, "Invalid from date")
+		}
+		fromDate = parsed
+	}
+
+	toDate := fromDate.AddDate(0, 0, 1)
+	if queryParams.ToDate != "" {
+		parsed, err := time.Parse(time.RFC3339, queryParams.ToDate)
+		if err != nil {
+			return fiber.NewError(fiber.StatusBadRequest, "Invalid to date")
+		}
+		toDate = parsed
+	}
+
+	if toDate.Before(fromDate) {
+		return fiber.NewError(fiber.StatusBadRequest, "To date must be after from date")
+	}
+
+	// Semi-arbitrary, but meant to prevent huge queries across long time spans
+	if toDate.Sub(fromDate) > 7*24*time.Hour {
+		return fiber.NewError(fiber.StatusBadRequest, "Time range must be <= 7 days")
+	}
+
+	sql := `
+	with leaders as (SELECT
+		user_change.owner,
+		user_change.account,
+		SUM(ABS(user_change.change)) / 100000000 AS volume
+	FROM sol_token_account_balance_changes user_change
+	JOIN sol_token_account_balance_changes vault_change
+		ON vault_change.signature = user_change.signature
+		AND vault_change.change = 0 - user_change.change
+		AND vault_change.mint = user_change.mint
+	JOIN sol_meteora_dbc_pools dbc_pool
+		ON dbc_pool.quote_vault = vault_change.account
+	WHERE
+		user_change.owner != 'FhVo3mqL8PW5pH5U2CN4XE33DokiyZnUwuGpH2hmHLuM'
```

Comment:
confirming these addresses for staging too / across all token deploys. maybe we can comment on what they are (meteora pool authorities dbc and damm right?)

Reply:
Will add a comment. I don't follow the other part. Are you saying there is another pair of addresses to add here or that these are used in both environments?

Reply:
^ This is what I have done :pray: